src/main/kotlin/platform/architectury/version/ArchitecturyVersion.kt

```diff
@@ -0,0 +1,144 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2021 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.platform.architectury.version
+
+import com.demonwav.mcdev.util.SemanticVersion
+import com.intellij.util.castSafelyTo
+import com.jetbrains.rd.util.first
+import java.io.IOException
+import java.net.URL
+import javax.xml.stream.XMLInputFactory
+import javax.xml.stream.events.XMLEvent
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
+
+class ArchitecturyVersion private constructor(val versions: Map<SemanticVersion, List<SemanticVersion>>) {
+
+    fun getArchitecturyVersions(mcVersion: SemanticVersion): List<SemanticVersion> {
+        val roundedVersion = Json.parseToJsonElement(
+            URL(
+                "https://gist.githubusercontent.com" +
+                    "/shedaniel/4a37f350a6e49545347cb798dbfa72b3" +
+                    "/raw/architectury.json"
+            ).readText()
+        ).jsonObject["versions"]!!.jsonObject.map { SemanticVersion.parse(it.key) }
```

Comment:
If the format changes, we should ideally fail gracefully rather than asserting things are not null.

Reply:
So do a chain of `?.` thingies and check if null at the end?